dcpy/utils/versions.py

```diff
@@ -125,7 +127,11 @@ def label(self) -> str:
     def __lt__(self, other) -> bool:
         match other:
             case Date():
-                return (self.date, self.format) < (other.date, other.format)
+                return (self.date, self.format, self.patch) < (
```

Comment:
question/nit: why is the format included in the ordering?

Reply:
not strictly about this PR, but glancing through... it looks like there's one case where we allow comparisons between classes that aren't the same, and that case (Date < MajorMinor) is unintuitive. I think we could chop about ~100 lines and a lot of complexity by just relying on dataclass sort orders. Thoughts? cc @fvankrieken @damonmcc

Reply:
To the nit - just to have something deterministic in the case of a "tie". Maybe a bit silly but I figured might as well

To the other question - so that code doesn't break when we moved FacDB from 2023-XX-XX to 24v1. I agree we generally don't want to be making those sorts of comparisons, but occasionally we do when we're changing versions to better align. I think any changes to this should happen more in context of our versioning discussion and outside of Sasha's PR